src/models/converters/safes.rs

```diff
@@ -49,22 +58,30 @@ impl SafeInfo {
 
 pub(super) fn calculate_version_state(
     safe_version: &str,
-    min_chain_version: &Option<String>,
+    supported_master_copies: &Vec<MasterCopy>,
+    min_chain_version: String,
 ) -> ImplementationVersionState {
-    if let Some(min_chain_version) = min_chain_version {
-        let sem_ver_safe = Version::parse(safe_version);
-        let sem_ver_min = Version::parse(min_chain_version);
+    let sem_ver_safe = Version::parse(safe_version);
+    let sem_ver_min = Version::parse(&min_chain_version);
 
-        if sem_ver_min.is_err() || sem_ver_safe.is_err() {
-            return ImplementationVersionState::Unknown;
-        }
+    let supported_versions = supported_master_copies
```

Comment:
We check supported mastercopies by address, not by version

Reply:
Does this mean that the config service should return the address as well? We only get the stringyfied semver.

Reply:
ah also for `1.3.0` and `1.3.0+L2` we would have different addresses, right? In that case the semver check passes but the addresses wouldn't match

Reply:
I will add unit tests to this and move this PR back to draft